File: Remove_Duplicates_from_Sorted_List_83.java
package leetcode;
/**
 * Created by Kevin Lee on 4/4/23
 */
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Remove_Duplicates_from_Sorted_List_83 {
    public ListNode deleteDuplicates(ListNode head) {
        if (head == null || head.next == null) return head;

        ListNode slow = head;
        ListNode fast = head.next;

        while (fast != null) {
            if (slow.val == fast.val) {
                slow.next = fast.next;
            } else {
                slow = slow.next;
            }
            fast = fast.next;
        }

        return head;
    }
}

File: JavaScript/Remove_Duplicates_from_Sorted_List_83.js
/**
 * Definition for singly-linked list.
 * function ListNode(val, next) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.next = (next===undefined ? null : next)
 * }
 */
/**
 * Method:
 *  Use two node: slow and fast
 */
var deleteDuplicates = function(head) {
    if (head == null || head.next == null) return head;

    let slow = head;
    let fast = head.next;
    while (fast != null) {
        if (slow.val === fast.val) {
            // to delete the node with same val.
            slow.next = fast.next;
        } else {
            slow = slow.next;
        }
        fast = fast.next;
    }

    return head;
};